src/industry_gui.cpp

```diff
@@ -578,9 +592,35 @@ class BuildIndustryWindow : public Window {
 		}
 	}
 
+	static void AskRemoveAllIndustriesCallback(Window *w, bool confirmed)
+	{
+		if (confirmed) {
+			const int numberOfIndustries = Industry::GetPoolSize();
+			for (int i = 0; i < numberOfIndustries; i++) {
+				auto* industry = Industry::Get(i);
+				if (industry) delete industry;
+			}
```

Comment:
```suggestion
			for (Industry *i : Industry::Iterate()) delete i;
```

Reply:
I must have overlooked the Iterate method, this is much cleaner indeed, thanks!